napari/layers/image/_async_image.py

```diff
@@ -586,101 +559,107 @@ def _set_view_slice(self):
 
-        # load_chunk() will return the request if the load was synchronous,
-        # otherwise it till return None meaning the load is in progress.
+        # Load the chunk(s), it could be sync or async.
         satisfied_request = self._slice.load_chunk(request)
 
-        if satisfied_request is not None:
+        if satisfied_request is None:
+            # A load is underway, we are no longer in the loaded state.
+            self.events.loaded()
+        else:
+            # The load was sync so use the chunk now.
             self.chunk_loaded(satisfied_request)
 
     def chunk_loaded(self, request: ChunkRequest) -> None:
         """The given ChunkRequest was satisfied, we can use the data now.
 
+        This routine is called synchronously from _load_async() above, or
+        it is called asynchronously sometime after the ChunkLoader finishes
+        loading the data in a worker.
+
         Parameters
         ----------
         request : ChunkRequest
             The request that was satisfied/loaded.
         """
-        # Ultimately this check should not be needed, but for now.
+        # Ultimately this check should not be needed, but for now to be sure.
         if request.data_id != id(self.data):
             LOGGER.warn(
                 "Loaded data_id=%d expected %d", request.data_id, id(self.data)
             )
             return  # was not for us
 
+        request.transpose_chunks(self._get_order())
+
         image = request.chunks['image']
 
-        # ASYNC_TODO: Complain if the loaded data's shape's rgb status was
-        # different from what we inferred it to be on load. Basically if
-        # the user lied about the shape of their array. Not sure if we want
-        # to handle this better somehow?
+        # ASYNC_TODO: If the shape is wrong relative to rgb it means the user
+        # liked to Dask about the shape of their array. When you create a
+        # Dask array you tell it the same and you could be wrong.
         if self.rgb != guess_rgb(image.shape):
             raise RuntimeError("Loaded chunk was the wrong shape.")
 
         # Tell the slice its data is ready to show.
         if not self._slice.chunk_loaded(request):
-            return  # was the stale/wrong chunk?
+            # ImageSlice rejected the chunk, it was for the wrong indices?
+            return
 
         if self.multiscale:
             self.events.scale()
             self.events.translate()
 
-        # Notify the world our loaded status changed.
+        # We are now in the loaded state.
         self.events.loaded()
 
         # Update vispy, draw the new slice
         self.events.set_data()
 
     def _update_thumbnail(self):
         """Update thumbnail with current image data and colormap."""
-        if self._slice is None or not self._slice.loaded:
+        # ASYNC_TODO: no point in computing the thumbnail until we've
+        # loaded, but do we need to have teh caller not call this until
```

Comment:
```suggestion
        # loaded, but do we need to have the caller not call this until
```

Reply:
That comment/typo when away in latest `image.py`.